pyomo/contrib/doe/fim_doe.py

```diff
@@ -0,0 +1,2582 @@
+                    for t in self.flatten_measure_timeset[j]:
+                        measure_string_name = self.measure.SP_measure_name(j,t,mode='sequential_finite')
+                        C_value = pyo.value(eval(measure_string_name))
+                        output_iter.append(C_value)
+
+                output_record[no_s] = output_iter
+
+            output_record['design'] = self.design_values
+            if store_output is not None:
+                f = open(store_output, 'wb')
+                pickle.dump(output_record, f)
+                f.close()
+
+            # calculate jacobian
+            jac = self.__finite_calculation(output_record, scena_gen)
+
+            time11 = time.time()
+            self.logger.info('Build time with sequential_finite mode [s]:', sum(time_allbuild))
+            self.logger.info('Solve time with sequential_finite mode [s]:', sum(time_allsolve))
+            self.logger.info('Total wall clock time [s]:', time11-time00)
+
+            # return all models formed
+            self.models = models
+
+        # Assemble and analyze results
+        if self.specified_prior is None:
+            prior_in_use = self.prior_FIM
+        else:
+            prior_in_use = self.specified_prior
+
+        FIM_analysis = FIM_result(self.param_name, self.measure, jacobian_info=None, all_jacobian_info=jac,
+                                    prior_FIM=prior_in_use, store_FIM=self.FIM_store_name, scale_constant_value=self.scale_constant_value)
+
+        # Store the Jacobian information for access by users
+        self.jac = jac
+
+        if read_output is None:
+            FIM_analysis.build_time = sum(time_allbuild)
+            FIM_analysis.solve_time = sum(time_allsolve)
+
+        return FIM_analysis
+
+    def __sequential_sipopt(self,read_output):
+        time00 = time.time()
+        # create scenario class for a base case
+        scena_gen = Scenario_generator(self.param_init, formula=None, step=self.step)
+        scenario_all = scena_gen.simultaneous_scenario()
+
+        # sipopt only uses backward difference scheme
+        # store measurements for scenarios
+        all_perturb_measure = []
+        all_base_measure = []
+        # store jacobian info
+        jac={}
+
+        # if measurements are provided
+        if read_output is not None:
+            with open(read_output, 'rb') as f:
+                output_record = pickle.load(f)
+                f.close()
+            jac = self.__finite_calculation(output_record, scena_gen)
+
+        else:
+            # time building time and solving time store list
+            time_allbuild = []
+            time_allsolve = []
+            # loop over parameters
+            for pa in range(len(self.param_name)):
+                perturb_mea = []
+                base_mea = []
+
+                # create model
+                time0_build = time.time()
+                mod = self.create_model(scenario_all, self.args)
```

Comment:
Do you have to recreate the model each time in this loop, or can you re-use the previous model and update it?